src/main/java/donnafin/logic/PersonAdapter.java

```diff
@@ -150,67 +169,116 @@ private Person editPersonTags(Person personToEdit, String newValue) {
+                personToEdit.getName(),
+                personToEdit.getPhone(),
+                personToEdit.getEmail(),
+                personToEdit.getAddress(),
+                newTags,
+                personToEdit.getNotes(),
                 personToEdit.getPolicies(),
-                personToEdit.getLiability(), personToEdit.getCommission(),
-                personToEdit.getAssetSet());
+                personToEdit.getLiabilities(),
+                personToEdit.getAssets());
     }
 
     private Person editPersonNotes(Person personToEdit, String newValue) {
         Notes newNotes = new Notes(newValue);
+
         return new Person(
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), personToEdit.getTags(), newNotes,
+                personToEdit.getName(),
+                personToEdit.getPhone(),
+                personToEdit.getEmail(),
+                personToEdit.getAddress(),
+                personToEdit.getTags(),
+                newNotes,
                 personToEdit.getPolicies(),
-                personToEdit.getLiability(), personToEdit.getCommission(),
-                personToEdit.getAssetSet());
+                personToEdit.getLiabilities(),
+                personToEdit.getAssets());
+    }
+
+    private Policy stringToPolicy(String policyString) {
+        String[] details = policyString.split(Attribute.DELIMITER);
+        //@parserteam do error handling for this
+        //        if (details != 5) {
+        //            throw new ParseException(Policy.MESSAGE_CONSTRAINTS);
+        //        }
+
+        return new Policy(details[0], details[1], details[2], details[3], details[4]);
     }
 
     private Person editPersonPolicies(Person personToEdit, String newValue) {
-        String[] policiesString = newValue.split(" ");
-        Set<Policy> newPolicies = Arrays.stream(policiesString)
-                .map(Policy::new)
+        String[] policiesStrings = newValue.split(SET_DELIMITER);
+        Set<Policy> newPolicies = Arrays.stream(policiesStrings)
+                .map(this::stringToPolicy)
                 .collect(Collectors.toSet());
+
         return new Person(
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), personToEdit.getTags(), personToEdit.getNotes(), newPolicies,
-                personToEdit.getLiability(), personToEdit.getCommission(),
-                personToEdit.getAssetSet());
+                personToEdit.getName(),
+                personToEdit.getPhone(),
+                personToEdit.getEmail(),
+                personToEdit.getAddress(),
+                personToEdit.getTags(),
+                personToEdit.getNotes(),
+                newPolicies,
+                personToEdit.getLiabilities(),
+                personToEdit.getAssets());
+    }
```

Comment:
I think this one need to think a bit, can't be the person has to retype all policies... i thought we decided to have delete / add / edit functions for Policy Liab Assets?

Reply:
handled by the parser team during edit